mlir/include/mlir/Dialect/OpenACC/OpenACCOps.td

```diff
@@ -989,7 +1008,8 @@ def OpenACC_ParallelOp : OpenACC_Op<"parallel",
 
   let assemblyFormat = [{
     oilist(
-        `dataOperands` `(` $dataClauseOperands `:` type($dataClauseOperands) `)`
+        `combined` `(` custom<CombinedConstructs>($combined) `)`
```

Comment:
I think we can make the IR less verbose especially for compute operations since they only combined with loop. If you use a UnitAttr, you don't even need a custom parser/printer here. 

So the IR would look like:

```
acc.parallel combined ... // we (omit parallel loop) since this is obvious. 
```

It will only print the keyword if the UnitAttr is present (I'm not even sure you need the ()?). 
```
(`combined` $combined)?
```

I would do the same for serial and kernels. 


Also you probably don't want this in the oilist.

Reply:
So I considered this at first - but dismissed this option. I preferred consistency in obtaining combined construct without any hidden semantics. This means one can query the compute and loop ops in an uniform manner and get the same attribute type value without having to do any additional interpretation.

And thus I extended the uniformity to the printing also - even though you are right that some information can be omitted.

Reply:
I think it makes sense to look for uniforming when querying the information but we should probably to that through an interface so it is robust. I actually would like to introduce some interfaces for couple of things that some operations share so that we can enforce uniformity. We might leave it for another patch. 

I still find the UnitAttr approach more elegant (quick bool query + no custom parser/printer)  but it's not a blocker if you don't like it.

Reply:
I used UnitAttr as requested :)